bson/bsonrw/extjson_parser.go

```diff
@@ -264,18 +264,60 @@ func (ejp *extJSONParser) readValue(t bsontype.Type) (*extJSONValue, error) {
 			return nil, err
 		}
 
-		// read KV pairs
-		keys, vals, err := ejp.readObject(2, false)
-		if err != nil {
-			return nil, err
-		}
-
 		ejp.advanceState()
-		if ejp.s != jpsSawEndObject {
-			return nil, invalidJSONErrorForType("2 key-value pairs and then }", t)
-		}
+		if t == bsontype.Binary && ejp.s == jpsSawValue {
```

Comment:
Instead of making this an if/else statement, can you just create a separate `case` statement for `bsontype.Binary`? Alternatively, just put a break at the end of the if statement and eliminate the else, as per [indent error flow](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow).

Reply:
I agree with Kris' first suggestion--`bsontype.Binary` should have its own `case` statement now, since it requires special handling. Though that would require something like this:
```
if ejp.s == jpsSawValue {
  ...
  break
}
fallthrough
```
followed by the regex, timestamp, and dbpointer `case ` to handle non-legacy binary. Not sure if that's ideal, style-wise.

Reply:
I don't see any way to do the `fallthrough` trick because the branching needs occur after reading past the colon.  After falling through, it would execute the code to read past the colon a second time.  That said, I modified it to eliminate the `else` block and use `break` instead.